restapi/datasource_api_object.go

```diff
@@ -115,34 +131,75 @@ func dataSourceRestApiRead(d *schema.ResourceData, meta interface{}) error {
     } /* End Loop through parts */
 
     data_array = (*ptr).([]interface{})
-  } else {
+  } else if "" == results_key && !single_obj {
     data_array = result.([]interface{})
+  } else { // single_obj
+    ptr := &result
+    single_object_data = (*ptr).(map[string]interface{})
   }
 
-  /* Loop through all of the results seeking the specific record */
-  for _, item := range data_array {
-    hash := item.(map[string]interface{})
-
-    /* We found our record */
-    if hash[search_key] == search_val {
-      id = hash[id_attribute].(string)
+  if !single_obj {
+
+    /* Loop through all of the results seeking the specific record */
+    for _, item := range data_array {
+      hash := item.(map[string]interface{})
+
+      // Parse search_key
+      search_parts := strings.Split(search_key, "/")
+      search_part := ""
+      search_seen := ""
+      search_hash := hash
+
+      // Loop through search parts
+      for len(search_parts) > 1 {
+        log.Printf("datasource_api_object.go: Looping through search_parts\n")
+        search_part, search_parts = search_parts[0], search_parts[1:]
+
+        // Protect against double slashes by mistake
+        if "" == search_part { break }
+
+        if _, ok := search_hash[search_part]; ok {
+          log.Printf("search_hash[search_part] exists: '%s'\n", search_hash[search_part])
+          search_hash = search_hash[search_part].(map[string]interface{})
+          search_seen += "/" + search_part
+        } else {
+          log.Printf("Failed to find %s in returned data structure after finding '%s'", search_part, search_seen)
```

Comment:
Should this be an error or invoke some sort of abort sequence? By not doing so, I guess this area of the code tolerates objects that don't have the full depth of `search_key` like `path/path2` when users are requesting `path/path2/path3/value`.

My initial thought is that this may be too lenient when finding an id... each object returned in the array should probably have the full depth of `search_value`.

Reply:
You're correct.  I think I originally did have return instead of log, but then changed it for some reason which is no longer relevant.  I just fixed that in new commit.